base/range.jl

```diff
@@ -801,3 +801,5 @@ end
 
 in{T<:Integer}(x, r::Range{T}) = isinteger(x) && !isempty(r) && x>=minimum(r) && x<=maximum(r) && (mod(convert(T,x),step(r))-mod(first(r),step(r)) == 0)
 in(x::Char, r::Range{Char}) = !isempty(r) && x >= minimum(r) && x <= maximum(r) && (mod(Int(x) - Int(first(r)), step(r)) == 0)
+
+unique{T}(x::Range{T}) = step(x) == zero(T) ? x[1:1] : x
```

Comment:
Perhaps the solution is to define a different method for type (LinRange). x[1] == x[end] should ordinarily be sufficient.